presto-hive/src/main/java/io/prestosql/plugin/hive/util/ResumableTasks.java

```diff
@@ -25,30 +27,33 @@ private ResumableTasks()
     {
     }
 
-    public static void submit(Executor executor, ResumableTask task)
+    public static ListenableFuture<Void> submit(Executor executor, ResumableTask task)
     {
+        SettableFuture<Void> completionFuture = SettableFuture.create();
         executor.execute(new Runnable()
         {
             @Override
             public void run()
             {
-                ResumableTask.TaskStatus status = safeProcessTask(task);
-                if (!status.isFinished()) {
-                    // if task is not complete, schedule it it to run again when the future finishes
-                    status.getContinuationFuture().addListener(this, executor);
+                ResumableTask.TaskStatus status;
+                try {
+                    status = task.process();
                 }
+                catch (Throwable t) {
+                    log.warn(t, "ResumableTask completed exceptionally");
+                    completionFuture.setException(t);
```

Comment:
should we also set `ResumableTask.TaskStatus.finished();`

Reply:
here, we act upon `TaskStatus`, but we do not return it